File: backend/src/main/java/com/back/fairytale/domain/thoughts/repository/ThoughtsRepository.java
package com.back.fairytale.domain.thoughts.repository;

import com.back.fairytale.domain.thoughts.entity.Thoughts;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface ThoughtsRepository extends JpaRepository<Thoughts, Long> {
    Optional<Thoughts> findByFairytaleIdAndUserId(Long fairytaleId, Long userId);
}

File: backend/src/main/java/com/back/fairytale/domain/thoughts/repository/ThoughtsRepository.kt
package com.back.fairytale.domain.thoughts.repository

import com.back.fairytale.domain.thoughts.entity.Thoughts
import org.springframework.data.jpa.repository.JpaRepository

interface ThoughtsRepository : JpaRepository<Thoughts, Long> {
    fun findByFairytaleIdAndUserId(fairytaleId: Long, userId: Long): Thoughts?
}
